File: backend/app/schemas/evidence.py
from pydantic import BaseModel
from typing import Optional, Dict, Any, List
import datetime

class EvidenceResponse(BaseModel):
    id: str
    ticket_id: str
    evidence_type: str
    source_type: str
    file_path: str
    file_type: str
    sha256_hash: str
    perceptual_hash: Optional[str] = None
    captured_at: Optional[datetime.datetime] = None
    uploaded_at: datetime.datetime
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    width: Optional[int] = None
    height: Optional[int] = None
    file_size_bytes: Optional[int] = None
    verification_session_id: Optional[str] = None

    class Config:
        from_attributes = True

class EvidenceVerifyRequest(BaseModel):
    evidence_id: str

class EvidenceVerifyResponse(BaseModel):
    status: str
    integrity_score: float
    details: Optional[Dict[str, Any]] = None

# Aliases for backwards compatibility
EvidenceUploadResponse = EvidenceResponse
VerificationResponse = EvidenceVerifyResponse

File: frontend/src/types/evidence.ts
export type EvidenceType = 'BEFORE' | 'AFTER' | 'LIVE_VERIFICATION';
export type SourceType = 'UPLOAD' | 'LIVE_CAMERA';

export interface EvidenceItem {
  id: string;
  ticket_id: string;
  evidence_type: EvidenceType;
  source_type: SourceType;
  file_path: string;
  file_type: string;
  sha256_hash: string;
  perceptual_hash?: string;
  captured_at?: string;
  uploaded_at: string;
  latitude?: number;
  longitude?: number;
  width?: number;
  height?: number;
  file_size_bytes?: number;
  verification_session_id?: string;
}

export interface UploadEvidencePayload {
  ticket_id: string;
  file: File;
  evidence_type: EvidenceType;
  source_type: SourceType;
  latitude?: number;
  longitude?: number;
}
